fxevent/zap.go

```diff
@@ -24,145 +24,175 @@ import (
 	"strings"
 
 	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
 // ZapLogger is an Fx event logger that logs events to Zap.
 type ZapLogger struct {
 	Logger *zap.Logger
+
+	logLevel   *zapcore.Level
+	errorLevel *zapcore.Level
 }
 
 var _ Logger = (*ZapLogger)(nil)
 
+// UseErrorLevel sets the level of error logs emitted by Fx to level.
+func (l *ZapLogger) UseErrorLevel(level zapcore.Level) {
+	l.errorLevel = &level
+}
+
+// UseLogLevel sets the level of non-error logs emitted by Fx to level.
+func (l *ZapLogger) UseLogLevel(level zapcore.Level) {
+	l.logLevel = &level
+}
+
+func (l *ZapLogger) logEvent(msg string, fields ...zap.Field) {
+	if l.logLevel == nil {
+		l.Logger.Log(zapcore.InfoLevel, msg, fields...)
+	} else {
+		l.Logger.Log(*l.logLevel, msg, fields...)
+	}
```

Comment:
For non-error, with logLevel being a non-ptr, this can just be:
```suggestion
	l.Logger.Log(l.logLevel, msg, fields...)
```